pkg/apis/logging/v1/input_types.go

```diff
@@ -0,0 +1,44 @@
+package v1
+
+import (
+	"github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1/inputs"
+	sets "k8s.io/apimachinery/pkg/util/sets"
+)
+
+// Reserved input names.
+const (
+	InputNameApplication    = "application"    // Non-infrastructure container logs.
+	InputNameInfrastructure = "infrastructure" // Infrastructure containers and system logs.
+	InputNameAudit          = "audit"          // System audit logs.
+)
+
+var ReservedInputNames = sets.NewString(InputNameApplication, InputNameInfrastructure, InputNameAudit)
+
+// InputSpec defines a source of log messages.
+type InputSpec struct {
+	// Name used to refer to the input of a `pipeline`.
+	//
+	// +required
+	Name string `json:"name"`
+
+	// Type of input source.
+	//
+	// +kubebuilder:validation:Enum:=application;infrastructure;audit
+	// +required
+	Type string `json:"type"`
+
+	// InputTypeSpec is inlined with a required `type` and optional extra configuration.
+	//
+	// +optional
+	InputTypeSpec `json:",inline,omitempty"`
+}
+
+// InputTypeSpec is a union of optional type-specific extra specs.
+//
+type InputTypeSpec struct {
+	// Filter for application logs.
+	// +optional
+	Application *inputs.ApplicationType `json:"application,omitempty"`
+
+	// TODO(alanconway) in future we may add infra, audit or other types of filter.
```

Comment:
this comment is a little confusing to me... do we support infra and audit as well?

Reply:
It is poorly worded. The idea is that we have an input selection spec that is specific to Applicaiton logs today, and uses namespaces. In future we may add other types of of selectors that are specific to infra/audit and use some other criteria. That's why the Application is separated as a "Type" to anticipate having other types in future, even though there aren't any yet.

Reply:
It is poorly worded. Today we only support selecting by namespace for application logs. In future we _may_ support filtering infra and/or audit logs by other criteria (namespace doesn't make much sense) For future-proofing we split out the ApplicationType as if there was going to be an enum with InfraType etc. - even though there isn't yet.